src/algebraic_topology/split_simplicial_object.lean

```diff
@@ -0,0 +1,227 @@
+with the direct sum of objects `s.N i` indexed by all possible
+epimorphisms `[n] ⟶ [i]` in `simplex_category`. (We do not
+assume that the morphisms `s.ι n` are monomorphisms: in the
+most common categories, this would be a consequence of the
+axioms.)
+
+## References
+* [Stacks: Splitting simplicial objects] https://stacks.math.columbia.edu/tag/017O
+
+-/
+
+noncomputable theory
+
+open category_theory
+open category_theory.category
+open category_theory.limits
+open opposite
+open_locale simplicial
+
+universe u
+
+variables {C : Type*} [category C]
+
+namespace simplicial_object
+
+namespace splitting
+
+/-- The index set which appears in the definition of split simplicial objects. -/
+def index_set (Δ : simplex_categoryᵒᵖ) :=
+Σ (Δ' : simplex_categoryᵒᵖ), { α : Δ.unop ⟶ Δ'.unop // epi α }
+
+namespace index_set
+
+/-- The element in `splitting.index_set Δ` attached to an epimorphism `f : Δ ⟶ Δ'`. -/
+@[simps]
+def mk {Δ Δ' : simplex_category} (f : Δ ⟶ Δ') [epi f] : index_set (op Δ) :=
+⟨op Δ', f, infer_instance⟩
+
+variables {Δ' Δ : simplex_categoryᵒᵖ} (A : index_set Δ)
+
+/-- The epimorphism in `simplex_category` associated to `A : splitting.index_set Δ` -/
+def e := A.2.1
+
+instance : epi A.e := A.2.2
+
+lemma ext' : A = ⟨A.1, ⟨A.e, A.2.2⟩⟩ := by tidy
+
+lemma ext (A₁ A₂ : index_set Δ) (h₁ : A₁.1 = A₂.1)
+  (h₂ : A₁.e ≫ eq_to_hom (by rw h₁) = A₂.e) : A₁ = A₂ :=
+begin
+  rcases A₁ with ⟨Δ₁, ⟨α₁, hα₁⟩⟩,
+  rcases A₂ with ⟨Δ₂, ⟨α₂, hα₂⟩⟩,
+  simp only at h₁,
+  subst h₁,
+  simp only [eq_to_hom_refl, comp_id, index_set.e] at h₂,
+  simp only [h₂],
+end
+
+instance : fintype (index_set Δ) :=
+fintype.of_injective
+  ((λ A, ⟨⟨A.1.unop.len, nat.lt_succ_iff.mpr
+    (simplex_category.len_le_of_epi (infer_instance : epi A.e))⟩, A.e.to_order_hom⟩) :
+    index_set Δ → (sigma (λ (k : fin (Δ.unop.len+1)), (fin (Δ.unop.len+1) → fin (k+1)))))
+begin
+  rintros ⟨Δ₁, α₁⟩ ⟨Δ₂, α₂⟩ h₁,
+  induction Δ₁ using opposite.rec,
+  induction Δ₂ using opposite.rec,
+  simp only at h₁,
+  have h₂ : Δ₁ = Δ₂ := by { ext1, simpa only [subtype.mk_eq_mk] using h₁.1, },
+  subst h₂,
+  refine ext _ _ rfl _,
+  ext : 2,
+  exact eq_of_heq h₁.2,
+end
+
+variable (Δ)
+
+/-- The distinguished element in `splitting.index_set Δ` which corresponds to the
+identity of `Δ`. -/
+def id : index_set Δ := ⟨Δ, ⟨𝟙 _, by apply_instance,⟩⟩
+
+instance : inhabited (index_set Δ) := ⟨id Δ⟩
+
+end index_set
+
+variables (N : ℕ → C) (Δ : simplex_categoryᵒᵖ)
+  (X : simplicial_object C) (φ : Π n, N n ⟶ X _[n])
+
+/-- Given a sequences of objects `N : ℕ → C` in a category `C`, this is
+a family of objects indexed by the elements `A : splitting.index_set Δ`.
+The `Δ`-simplices of a split simplicial objects shall identify to the
+direct sum of objects in such a family. -/
+@[simp, nolint unused_arguments]
+def summand (A : index_set Δ) : C := N A.1.unop.len
+
+variable [has_finite_coproducts C]
+
+/-- The direct sum of the family `summand N Δ` -/
+@[simp]
+def sum := sigma_obj (summand N Δ)
```

Comment:
Very minor, but why not use the notation for coproducts?

Reply:
Ok. I have made this change.